aws/resource_aws_elasticache_cluster.go

```diff
@@ -268,32 +275,37 @@ func resourceAwsElasticacheCluster() *schema.Resource {
 				return errors.New(`az_mode "cross-az" is not supported with num_cache_nodes = 1`)
 			},
 			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
```

Comment:
It would help with readability here to define all the functions and then list them here by name.

Reply:
Yeah, I've been thinking about that. It would also make them testable if needed

Reply:
It's also really easy to miss the one named function among all the inline functions

Reply:
I actually did that exact thing on my first read-through :-)